### PR TITLE
moved vcs.go and vcs_test.go to misc

### LINKED ISSUE #572: Abstract away VCS URI parsing for code deduplication
There are multiple places where version control URIs are parsed, they can probably be abstracted away into a common library.

For context, follow up on discussion for https://github.com/guacsec/guac/pull/557#discussion_r1132485546

### PR DESCRIPTION
**Pull request by arorasoham9**
# Description of the PR
moved vcs.go and vcs_test.go to misc

closes #572

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

**Review by pxp928 — COMMENTED** [review]
So there is more to that issue and not just moving the files. The github collector has duplicate code that needs to be removed and replaced by calling the VCS helper functions. See comment: https://github.com/guacsec/guac/pull/557#discussion_r1132485546

**Comment by arorasoham9** [comment]
@pxp928  Yes, didn't realize that earlier, I'll fix that right away.

**Comment by arorasoham9** [comment]
Still have one issue to fix, do not merge.

**Comment by arorasoham9** [comment]
@pxp928 Ready to merge.

**Review by pxp928 — COMMENTED** [review]
Looks good! Thanks! @mlieberman85 wanted your final approval on this PR as you wrote the GitHub collector.

File: Makefile [hunk 1]
```diff
@@ -40,7 +40,7 @@ cover: test
 
 # Check the formatting
 .PHONY: fmt
-fmt:
+fmt:
```
Comment (mihaimaruseac) [inline Makefile:43]:
```suggestion
fmt:
```

(minor nit: trailing whitespace)

**Review by mihaimaruseac — APPROVED** [review]

**Comment by arorasoham9** [comment]
@pxp928 fixed the makefile and added sign-offs.

File: Makefile [hunk 2]
```diff
@@ -5,7 +5,7 @@ PKG=github.com/guacsec/guac/pkg/version
 LDFLAGS="-X $(PKG).Version=$(VERSION) -X $(PKG).Commit=$(COMMIT) -X $(PKG).Date=$(BUILD)"
 
 CONTAINER ?= docker
-CPUTYPE=$(shell uname -m)
+CPUTYPE=$(shell uname -m | sed 's/x86_64/amd64/')
```
Comment (pxp928) [inline Makefile:8]: this looks like a rebase issue. Should not be part of your PR.

Comment (arorasoham9) [reply Makefile]: Corrected.

Comment (arorasoham9) [reply Makefile]: Would that help?

Comment (arorasoham9) [reply Makefile]: Ok. Should be fixed now.

**Review by pxp928 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**